### PR TITLE
Add last resolve time metric

### PR DESCRIPTION
**Pull request by lalalalatt**
As the description in https://github.com/authzed/spicedb/issues/422:

> It's important for an operator to know that spicedb is continually receiving updated peer lists. Emit metrics that can be used to alert if the list is not updated expediently.

### CODE REVIEW

**Comment by lalalalatt** [comment]
@sercand please review this PR, Thanks

File: builder.go [hunk 1]
```diff
@@ -283,6 +296,7 @@ func (k *kResolver) watch() error {
 			if hasMore {
 				k.handle(up.Object)
 			} else {
+				k.lastResolve.Set(float64(time.Since(k.lastResolveTime).Seconds()))
```
Comment (sercand) [inline builder.go:299]: @lalalalatt do you achieve what do you want with this change or did your test it?  I don't think that lastResolve will update frequently therefore what you will see is "0" on prometheus metrics.

**Comment by lalalalatt** [comment]
Thanks for the correction. 
I misunderstood the interaction between grpc resolver and the kResolver previously.

---

And I have add a test for the `resolve lag` metric, here is the full test result:

```
go test . -v
=== RUN   TestBuilder
0, address: 192.168.194.36:53
0, servername: kube-dns.kube-system
--- PASS: TestBuilder (0.02s)
=== RUN   TestResolveLag
0, address: 192.168.194.36:53
0, servername: kube-dns.kube-system
    builder_test.go:93: resolver lag: 5.000515125 s
--- PASS: TestResolveLag (6.00s)
=== RUN   TestParseResolverTarget
--- PASS: TestParseResolverTarget (0.00s)
=== RUN   TestParseTargets
--- PASS: TestParseTargets (0.00s)
PASS
ok      github.com/sercand/kuberesolver/v5      6.306s
```

**Comment by lalalalatt** [comment]
I change the metric to record the latest `resolverClient.UpdateState()` function call time to prevent the frequently trigger ticker added.
Then the user can set the observation alert by set some threshold on the `time.Now() - last update unix` value.

**Review by sercand — APPROVED** [review]

**Pull request merged**